server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMailboxesMethodTest.java

```diff
@@ -1154,6 +1158,273 @@ public void setMailboxesShouldReturnUpdatedWhenNameUpdateAskedOnExistingMailbox(
+                "      }" +
+                "    }," +
+                "    \"#0\"" +
+                "  ]" +
+                "]";
+        String requestBody2 =
+            "[" +
+                "  [ \"setMailboxes\"," +
+                "    {" +
+                "      \"update\": {" +
+                "        \"" + mailboxId + "\" : {" +
+                "          \"sharedWith\" : {}" +
+                "        }" +
+                "      }" +
+                "    }," +
+                "    \"#0\"" +
+                "  ]" +
+                "]";
+
+        with()
+            .header("Authorization", this.accessToken.serialize())
+            .body(requestBody1)
+            .post("/jmap");
+
+        with()
+            .header("Authorization", this.accessToken.serialize())
+            .body(requestBody2)
+            .post("/jmap");
+
+        given()
+            .header("Authorization", accessToken.serialize())
+            .body("[[\"getMailboxes\", {\"ids\": [\"" + mailboxId + "\"]}, \"#0\"]]")
+        .when()
+            .post("/jmap")
+        .then()
+            .statusCode(200)
+            .body(NAME, equalTo("mailboxes"))
+            .body(FIRST_MAILBOX + ".name", equalTo(myBox))
+            .body(FIRST_MAILBOX + ".sharedWith", is(ImmutableMap.of()));
+    }
+
+    @Test
+    public void updateShouldModifyStoredDataWhenSwithingACLUser() {
+        String myBox = "myBox";
+        String user1 = "user1";
+        String user2 = "user2";
+        jmapServer.getProbe(MailboxProbeImpl.class).createMailbox(MailboxConstants.USER_NAMESPACE, username, myBox);
+        Mailbox mailbox = jmapServer.getProbe(MailboxProbeImpl.class).getMailbox(MailboxConstants.USER_NAMESPACE, username, "myBox");
+        String mailboxId = mailbox.getMailboxId().serialize();
+        String requestBody1 =
+            "[" +
+                "  [ \"setMailboxes\"," +
+                "    {" +
+                "      \"update\": {" +
+                "        \"" + mailboxId + "\" : {" +
+                "          \"sharedWith\" : {\"" + user1 + "\": [\"a\", \"w\"]}" +
+                "        }" +
+                "      }" +
+                "    }," +
+                "    \"#0\"" +
+                "  ]" +
+                "]";
+        String requestBody2 =
+            "[" +
+                "  [ \"setMailboxes\"," +
+                "    {" +
+                "      \"update\": {" +
+                "        \"" + mailboxId + "\" : {" +
+                "          \"sharedWith\" : {\"" + user2 + "\": [\"a\", \"w\"]}" +
+                "        }" +
+                "      }" +
+                "    }," +
+                "    \"#0\"" +
+                "  ]" +
+                "]";
+
+        with()
+            .header("Authorization", this.accessToken.serialize())
+            .body(requestBody1)
+            .post("/jmap");
+
+        with()
+            .header("Authorization", this.accessToken.serialize())
```

Comment:
you can remove these this.

Reply:
Funny, I quickly investigated the same things.

No I can not! It's not part of the request definition (because you use request specification to query the authentication endpoint).

That would need a reallocation, and moreover is done nowhere else in JMAP integration tests. I chose concistency. Also thought "too much refactoring" in a single week.

Reply:
sorry I don't understand why this.accessToken is different from accessToken here?

Reply:
Ah yes of course I can remove this, not the access token.